tools/generators/README.md

```diff
@@ -0,0 +1,70 @@
+# `.xcodeproj` generators
+
+The `.xcodeproj` file bundle has multiple files inside of it. The generators
+in this package are responsible for creating those various files.
+
+## `PBXProj`
+
+The most important file in the `.xcodeproj` file bundle is the `project.pbxproj`
+file. This file details the project itself, and controls almost of all Xcode’s
+behavior related to the project. Other projects, such as
+[XcodeProj](https://github.com/tuist/XcodeProj), call the type that this file
+represents “`PBXProj`”, so we will too from here on.
+
+Instead of there being a single generator for `PBXProj`, there are multiple,
+with each generating a portion of the end file. These portions are called
+“partials”. These partials are concatenated together to form the final
+`project.pbxproj` file. We use this approach for a couple reasons:
+
+- Each generator only needs a subset of the total inputs needed to generate the
+  full `PBXProj`. This means that when the project changes, not every generator
+  will need to run, resulting in faster project generation.
+- Each generator can receive “raw” input, reducing the amount of Bazel analysis
+  phase CPU time and memory retained, resulting in faster project generation and
+  lower Bazel server constant memory usage.
+
+Below is a list of the `PBXProj` partial generators, and a high level
+description of what each one does. Please see their associated READMEs for more
+information.
+
+- `pbxproject_additional_elements`:
+  - The start of the `PBXProj` element
+  - All of the various `PBXProject` additional elements:
+    - `XCBuildConfiguration`
+    - `XCBuildConfigurationList`
+    - `PBXTargetDependency`
+    - `PBXContainerItemProxy`
+- [`pbxproject_prefix`](pbxproject_prefix/README.md):
+  - The start of `PBXProject` element
+  - Contains all `PBXProject` properties except for
```

Comment:
Nit: It was a bit confusing to see "all X except for Y" for something called "prefix", I'd expect a "prefix" to generate a piece (the start) of the thing and not almost "all" of it. This is explicitly stated in `pbxproject_prefix/README.md` and IMO a similar wording could be used here

Reply:
What different wording? They seem identical to me.

Reply:
My confusion was because I don't see the word "all" in`pbxproject_prefix/README.md`:
> The `pbxproject_prefix` generator creates a `PBXProj` partial containing the
start (i.e. prefix) of the `PBXProject` element.

But I'm overcomplicating this, this `README` here does mention `start` in the first item and then details what to expect specifically: all `PBXProject` except for X and Y

I'm going to close this nvm, not even a nit tbh haha